### PR TITLE
feat: 汎用 Colab ファインチューニングノートブック追加

### LINKED ISSUE #305: feat: あみたろ ITAコーパス ファインチューニングモデル (JA)
## Summary

あみたろの声素材工房が提供する ITAコーパス読み上げ音声を使用して、日本語ファインチューニングモデルを作成する。

## モチベーション

つくよみちゃん (100文) に続く2番目の日本語ファインチューニングモデル。アニメ声の女性話者で、複数スタイル (ノーマル/るんるん/よふかし/ぷんすか) が利用可能。商用利用可能なライセンスのため、ユーザーが安心して利用できる。

## データセット

| 項目 | 詳細 |
|------|------|
| 提供元 | [あみたろの声素材工房](https://amitaro.net/voice/corpus-list/ita/) |
| ライセンス | 独自利用規約（商用利用OK、AI音声合成モデルの学習を明示的に許可） |
| 発話数 | ITA 424文 × 4スタイル（ノーマル v2.1、るんるん v1.0、よふかし v1.1、ぷんすか v1.0） |
| クレジット | 必須（あみたろの声を使用していること + あみたろ本人が作成したものではないことを明記） |
| 制限事項 | センシティブな作品への使用不可（あみたろの声が25%以下のモデルなら用途制限なし） |

## タスク

- [ ] あみたろ ITAコーパス音声のダウンロード
- [ ] 音声前処理（22,050Hz リサンプリング、正規化）
- [ ] LJSpeech 形式への変換
- [ ] `prepare_multilingual_dataset.py` でデータセット作成
- [ ] 6lang ベースモデルからファインチューニング実行
- [ ] ONNX エクスポート + 推論テスト
- [ ] HuggingFace へのモデル公開
- [ ] ライセンス・クレジット表記の設定

## 技術メモ

- ベースモデル: 6lang 75 epoch (571話者)
- ファインチューニング: Template B (シングルスピーカー) に準拠
- `--resume-from-multispeaker-checkpoint` で転移学習
- つくよみちゃんと同様の手順で実施可能

### PR DESCRIPTION
**Pull request by ayutaz**
## Summary

- LJSpeech 形式のデータセットから Colab 上でシングルスピーカーファインチューニングを実行できる汎用ノートブックを追加
- 6lang ベースモデル (571話者, 75 epoch) からの転移学習を Template B 準拠で実施
- 6言語 (ja/en/zh/es/fr/pt) に対応、`SPEAKER_NAME` と `PRIMARY_LANGUAGE` を設定するだけで利用可能

## 内容

`notebooks/finetune.ipynb`:
1. 環境セットアップ (uv + 依存関係)
2. HuggingFace からベースモデルチェックポイント取得
3. LJSpeech 形式データセット (wavs/ + metadata.csv) のアップロード / Google Drive からの読み込み
4. 音声前処理 (22,050Hz リサンプリング、自動スキップ対応)
5. G2P 音素化 (`text_to_phoneme_ids_and_prosody`) → dataset.jsonl + config.json 生成
6. ファインチューニング実行 (500 epoch, lr=2e-5, freeze-dp 自動有効化)
7. ONNX エクスポート (EMA + FP16)
8. 推論テスト (言語に応じたテストテキスト) + Colab 上で音声再生
9. Google Drive へのモデル保存

## ユーザー設定項目

```python
SPEAKER_NAME = "my_speaker"   # 話者名 (出力パス・モデル名に使用)
PRIMARY_LANGUAGE = "ja"        # データセットの言語 (ja/en/zh/es/fr/pt)
SOURCE_METHOD = "upload"       # "upload" or "gdrive"
```

Closes #305

## Test plan

- [ ] Colab Free (T4) で環境セットアップが正常に完了すること
- [ ] LJSpeech 形式 ZIP のアップロード・自動検出が正しく動作すること
- [ ] dataset.jsonl の生成が正常に完了すること
- [ ] ファインチューニングが開始され、チェックポイントが保存されること
- [ ] ONNX エクスポート + 推論テストが動作すること

### CODE REVIEW

File: notebooks/amitaro_finetune.ipynb [hunk 1]
```diff
@@ -0,0 +1,874 @@
...
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!nvidia-smi\n",
+    "import torch\n",
+    "print(f\"CUDA available: {torch.cuda.is_available()}\")\n",
+    "if torch.cuda.is_available():\n",
+    "    print(f\"GPU: {torch.cuda.get_device_name(0)}\")\n",
+    "    print(f\"VRAM: {torch.cuda.get_device_properties(0).total_mem / 1024**3:.1f} GB\")"
```
[the remark this answers is not in the document]
Comment (ayutaz) [reply notebooks/amitaro_finetune.ipynb]: 修正しました。`total_memory` に変更済みです。

File: notebooks/amitaro_finetune.ipynb [hunk 2]
```diff
@@ -0,0 +1,874 @@
...
+   "source": [
+    "def encode_text(text, language=\"ja\"):\n",
+    "    \"\"\"Phonemize text and encode to phoneme IDs.\"\"\"\n",
+    "    phonemes = phonemizer.phonemize(text, language=language)\n",
+    "    phoneme_ids = phonemizer.encode(phonemes, id_map=id_map)\n",
+    "    return phoneme_ids, phonemes\n",
+    "\n",
+    "\n",
+    "def cache_audio(wav_path, cache_dir, sample_rate=SAMPLE_RATE):\n",
+    "    \"\"\"Cache normalized audio and spectrogram.\"\"\"\n",
```
[the remark this answers is not in the document]
Comment (ayutaz) [reply notebooks/amitaro_finetune.ipynb]: 修正しました。`text_to_phoneme_ids_and_prosody()` を使用し、BOS/EOS/intersperse padding が正しく適用されるようにしました。prosody_features も dataset.jsonl に含めるよう修正済みです。

File: notebooks/amitaro_finetune.ipynb [hunk 3]
```diff
@@ -0,0 +1,874 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import glob\n",
+    "\n",
+    "# 最新のチェックポイントを検索\n",
+    "ckpt_pattern = f\"{OUTPUT_DIR}/lightning_logs/version_*/checkpoints/epoch=*.ckpt\"\n",
+    "checkpoints = sorted(glob.glob(ckpt_pattern))\n",
```
[the remark this answers is not in the document]
Comment (ayutaz) [reply notebooks/amitaro_finetune.ipynb]: 修正しました。epoch 番号を正規表現で抽出してソートするようにしました。

File: notebooks/amitaro_finetune.ipynb [hunk 4]
```diff
@@ -0,0 +1,874 @@
...
+   "outputs": [],
+   "source": [
+    "# Colab で音声を再生\n",
+    "from IPython.display import Audio, display\n",
+    "import glob\n",
+    "\n",
+    "output_wavs = sorted(glob.glob(\"/content/test_output_*.wav\"))\n",
+    "for wav in output_wavs:\n",
+    "    print(f\"\\n{os.path.basename(wav)}:\")\n",
+    "    display(Audio(wav))"
```
[the remark this answers is not in the document]
Comment (ayutaz) [reply notebooks/amitaro_finetune.ipynb]: 修正しました。各テストの `{output_dir}/output.wav` を直接参照するように変更済みです。

File: notebooks/amitaro_finetune.ipynb [hunk 5]
```diff
@@ -0,0 +1,874 @@
...
+    "FILTER_LENGTH = 1024\n",
+    "WINDOW_LENGTH = 1024\n",
+    "HOP_LENGTH = 256\n",
+    "\n",
+    "os.makedirs(CACHE_DIR, exist_ok=True)\n",
+    "\n",
+    "# Phonemizer & ID map\n",
+    "phonemizer = MultilingualPhonemizer()\n",
+    "id_map = get_phoneme_id_map(LANGUAGE_KEY)\n",
+    "num_symbols = max(max(ids) for ids in id_map.values()) + 1\n",
```
[the remark this answers is not in the document]
Comment (ayutaz) [reply notebooks/amitaro_finetune.ipynb]: 修正しました。`MultilingualPhonemizer` の直接使用をやめ、プロジェクト標準の `text_to_phoneme_ids_and_prosody()` に置き換えました。

File: notebooks/amitaro_finetune.ipynb [hunk 6]
```diff
@@ -0,0 +1,874 @@
...
+    "if os.path.exists(ONNX_OUTPUT):\n",
+    "    for i, (text, lang) in enumerate(test_texts):\n",
+    "        output_wav = f\"/content/test_output_{i}_{lang}.wav\"\n",
+    "        !cd /content/piper-plus && CUDA_VISIBLE_DEVICES=\"\" uv run python -m piper_train.infer_onnx \\\n",
+    "            --model {ONNX_OUTPUT} \\\n",
+    "            --config {config_path} \\\n",
+    "            --output-dir /content \\\n",
+    "            --text \"{text}\" \\\n",
+    "            --language {LANGUAGE_KEY} --speaker-id 0 --noise-scale 0.667\n",
+    "        print(f\"Generated: {text[:30]}...\")\n",
```
[the remark this answers is not in the document]
Comment (ayutaz) [reply notebooks/amitaro_finetune.ipynb]: 修正しました。テストごとに別の output ディレクトリを使用するようにしました。

File: notebooks/amitaro_finetune.ipynb [hunk 7]
```diff
@@ -0,0 +1,874 @@
...
+    "\n",
+    "6言語マルチリンガルベースモデル (571話者, 75 epoch) から、あみたろの声素材工房 ITAコーパス音声を使った日本語ファインチューニングを行います。\n",
+    "\n",
+    "**ベースモデル:** `ayousanz/piper-plus-base` (6lang, epoch=74-step=504712)\n",
+    "\n",
+    "**データセット:** あみたろ ITAコーパス (424文 x 4スタイル = ~1,696発話)\n",
+    "\n",
+    "**ライセンス:** 商用利用OK (クレジット必須)\n",
+    "\n",
+    "[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/ayutaz/piper-plus/blob/feat/amitaro-colab-notebook/notebooks/amitaro_finetune.ipynb)"
```
[the remark this answers is not in the document]
Comment (ayutaz) [reply notebooks/amitaro_finetune.ipynb]: 修正しました。`dev` ブランチを指すように変更済みです。
